### PR TITLE
Better tristate support ddr5

### PR DESCRIPTION
**Pull request by mtdudek**
Better Tristate support in the DDR5 BasePHY. Updated SimPHY to correctly support tristate serialization.

### CODE REVIEW

File: litedram/phy/ddr5/s7phy.py [hunk 1]
```diff
@@ -44,11 +44,11 @@ def __init__(self, pads, *, iodelay_clk_freq, with_odelay, **kwargs):
...
         assert iodelay_clk_freq in [200e6, 300e6, 400e6]
         iodelay_tap_average = 1 / (2*32 * iodelay_clk_freq)
-        half_sys8x_taps = math.floor(self.tck / (4 * iodelay_tap_average))
-        assert half_sys8x_taps < 32, "Exceeded ODELAYE2 max value: {} >= 32".format(half_sys8x_taps)
+        half_sys4x_taps = math.floor(self.tck / (4 * iodelay_tap_average))
+        assert half_sys4x_taps < 32, "Exceeded ODELAYE2 max value: {} >= 32".format(half_sys4x_taps)
 
         # Registers --------------------------------------------------------------------------------
-        self._half_sys8x_taps = CSRStorage(5, reset=half_sys8x_taps)
+        self._half_sys8x_taps = CSRStorage(5, reset=half_sys4x_taps)
```
Comment (koluckirafal) [inline litedram/phy/ddr5/s7phy.py]: Mismatched variable name, shouldn't it be `self._half_sys4x_taps`?

File: litedram/phy/s7ddrphy.py [hunk 2]
```diff
@@ -389,10 +399,12 @@ def cdc(i):
         self.submodules += dq_oe_delay
         self.comb += dq_oe_delay.input.eq(dqs_preamble | dq_oe | dqs_postamble)
 
-        dq_dqs_ratio = databits // strobes
-        assert (dq_dqs_ratio in [4, 8])
-
         for i in range(databits):
+            __temp = Signal()
```
Comment (koluckirafal) [inline litedram/phy/s7ddrphy.py]: I know it's only a temporary signal, but I think it can be named more clearly to understand its purpose.

**Review by koluckirafal — COMMENTED** [review]
I left some small comments; other than that, it looks fine for me.

File: litedram/phy/ddr5/s7phy.py (hunk 1 above)
Comment (mtdudek) [reply litedram/phy/ddr5/s7phy.py]: This value was later removed in  PR #50, so I'll remove it now and rebase #50

File: litedram/phy/s7ddrphy.py (hunk 2 above)
Comment (mtdudek) [reply litedram/phy/s7ddrphy.py]: changed to __dq_sel

**Comment by tmichalak** [comment]
LGTM

**Pull request merged**